content/articles/fastify-svelte-app/index.md

```diff
@@ -0,0 +1,466 @@
+                                    <label for="title">Title:</label>
+                                    <input bind:value={newPostData.title}
+                                    type="text" class="form-control" id="text" placeholder="Note Post">
+
+                                </div>
+
+
+                                <button on:click|preventDefault={createPost}
+                                type="submit" class="btn btn-primary">Add</button>
+
+                            </form>
+                        </div>
+                    </div>
+                </div>
+
+                <div class="col-md-">
+                    <div class="card">
+                        <h2 class="card-header">Posts</h2>
+                        <div class="card-body">
+
+
+{#await fetchPosts}
+    <p>...waiting</p>
+
+{:then data}
+                  
+    <div>
+        {#each data as item}
+        <h3 class="card-title underline">{item.id}</h3>
+
+        <p class="card-text">
+            {item.title}
+            <span><button class="btn btn-danger">Delete</button></span>
+        </p>
+
+        {/each}
+
+    </div>
+    
+{:catch error}
+    <p>An error occurred!</p>
+
+{/await}
+                        </div>
+                    </div>
+                </div>
+
+            </div>
+        </div>
+    </div>
+</section>
+
+<style>
+    .underline{
+        text-decoration: underline;
+        color: rgb(44, 128, 255);
+    }
+    section{
+        display: flex;
+    }
+
+    .card-text{
+        display: flex;
+        align-items: center;
+    }
+    .card-text span{
+        margin-left: 50px;
+    }
+
+</style>
+```
+
+Let's briefly understand the above `Posts.svelte` code:
+
+Our `Posts.svelte` file has `<script>`, `<section>` container for our markup, and `<style>` blocks. In between the `<scipt>` tag,  we write the JavaScript code to add the dynamic functionality. To begin, we create a `newPostData` object and export it. It will hold the text of the new todo with a title attribute. The backend determines the `ID` by incrementing it each time we add a new post.
+To insert a new post, we create a function `createPost()` that sends a POST request to our backend in JSON format. To achieve this, we use the value from the `<input>` HTML element and bind it to the `newPostData`. Svelte enriches our HTML markup so that we can use programming logic such as conditionals and loops. If we click the Add button, the `createPost` method is invoked to create a new post and our backend handles the rest.
+
+Next, we need to understand how `fetchPosts` retrieves our posts. Data fetching is an asynchronous task. Our function and markup uses the async-await syntax to achieve this. The fetch API makes a network request and populates which returns our posts. On our markup, we check if `fetchPosts` method has retrieved the data before rendering it on the UI. Error handling is implemented on the catch block. The Html page:
```

Comment:
Break down the above 2 code snippets. And, explain them briefly under each smaller code snippet.